bin/mycroft-config

```diff
@@ -0,0 +1,238 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SOURCE="${BASH_SOURCE[0]}"
+cd -P "$( dirname "$SOURCE" )"
+DIR="$( pwd )"
+script=${0}
+script=${script##*/}
+
+function help() {
+    echo "${script}:  Mycroft configuration manager"
+    echo "usage: ${script} [COMMAND] [params]"
+    echo
+    echo "COMMANDs:"
+    echo "  edit (system|user)                  edit and validate config file"
+    echo "  reload                              instruct services to reload config"
+    echo "  show (default|remote|system|user)   display the specified setting file"
+    echo "  set <var>                           set the variable (under USER)"
+    echo "  get [var]                           display a particular variable"
+    echo "                                      or all if no 'var' specified"
+    echo "Note: Use jq format for specifying <var>"
+    echo
+    echo "Examples:"
+    echo "  ${script} edit user"
+    echo "  sudo ${script} edit system"
+    echo "  ${script} show remote"
+    echo "  ${script} get"
+    echo "  ${script} get enclosure.platform"
+    echo "  ${script} set test.subvalue \"foo\" "
+
+    exit 1
+}
+
+################################################################
+# Setup stuff based on the environment
+
+VIEWER="nano --syntax=json --view"
+if [ -z "$EDITOR" ] ; then
+    EDITOR="nano --syntax=json --tempfile"
+fi
+
+if [ -z "$TEMP" ] ; then
+    TEMP="/tmp"
+fi
+
+function found_exe() {
+    hash "$1" 2>/dev/null
+}
+
+if found_exe tput ; then
+    GREEN="$(tput setaf 2)"
+    BLUE="$(tput setaf 4)"
+    CYAN="$(tput setaf 6)"
+    YELLOW="$(tput setaf 3)"
+    RESET="$(tput sgr0)"
+    HIGHLIGHT=${YELLOW}
+fi
+
+################################################################
+# Utilities
+
+function validate_config_file() {
+    if [ ! -f "$1" ] ; then
+        # A missing config file is valid
+        return 0
+    fi
+
+    echo -n ${BLUE}
+
+    # Remove any comments (lines starting with # or //) found in the file and
+    # Use jq to validate and output errors
+    sed 's/^\s*[#\/].*$//g' "$1" | sed '/^$/d' | jq -e "." > /dev/null
+    result=$?
+
+    echo -n ${RESET}
+
+    #xxx echo "RESULT=$result for $1"
+    return $result
+}
+
+_conf_file="~/.mycroft/mycroft.conf"
+function name_to_path() {
+    case ${1} in
+        "system")     _conf_file="/etc/mycroft/mycroft.conf" ;;
+        "user")       _conf_file=$(readlink -f ~/.mycroft/mycroft.conf) ;;
+        "default")    _conf_file="$DIR/../mycroft/configuration/mycroft.conf" ;;
+        "remote")     _conf_file="/var/tmp/mycroft_web_cache.json" ;;
+
+        *)
+            echo "ERROR: Unknown name '${1}'."
+            echo "       Must be one of: default, remote, system, or user"
+            exit 1
+    esac
+}
+
+################################################################
+
+function edit-config() {
```

Comment:
Let's use underscores instead of hyphens to keep constency and portability

Reply:
Done